patches/server/0575-Configurable-door-breaking-difficulty.patch

```diff
@@ -5,58 +5,65 @@ Subject: [PATCH] Configurable door breaking difficulty
-+                .collect(Collectors.toList()),
-+            net.minecraft.world.Difficulty.class
-+        );
-+        vindicatorBreakDoors = getEnumList(
-+            "door-breaking-difficulty.vindicator",
-+            java.util.Arrays.stream(net.minecraft.world.Difficulty.values())
-+                .filter(net.minecraft.world.entity.monster.Vindicator.DOOR_BREAKING_PREDICATE)
-+                .collect(Collectors.toList()),
-+            net.minecraft.world.Difficulty.class
-+        );
++        for (final net.minecraft.world.entity.EntityType<?> entityType : new net.minecraft.world.entity.EntityType[]{net.minecraft.world.entity.EntityType.ZOMBIE, net.minecraft.world.entity.EntityType.ZOMBIE_VILLAGER, net.minecraft.world.entity.EntityType.ZOMBIFIED_PIGLIN, net.minecraft.world.entity.EntityType.HUSK, net.minecraft.world.entity.EntityType.VINDICATOR}) {
++            java.util.function.Predicate<net.minecraft.world.Difficulty> difficultyPredicate = net.minecraft.world.entity.monster.Zombie.DOOR_BREAKING_PREDICATE;
++            if (entityType == net.minecraft.world.entity.EntityType.VINDICATOR) {
++                difficultyPredicate = net.minecraft.world.entity.monster.Vindicator.DOOR_BREAKING_PREDICATE;
++            }
++            entitiesDifficultyBreakDoors.put(
++                entityType,
++                getEnumList(
++                    "door-breaking-difficulty.".concat(entityType.id),
++                    java.util.Arrays.stream(net.minecraft.world.Difficulty.values())
++                        .filter(difficultyPredicate)
++                        .collect(Collectors.toList()),
++                    net.minecraft.world.Difficulty.class
++                )
++            );
++        }
 +    }
 +
      public short keepLoadedRange;
      private void keepLoadedRange() {
          keepLoadedRange = (short) (getInt("keep-spawn-loaded-range", Math.min(spigotConfig.viewDistance, 10)) * 16);
 diff --git a/src/main/java/net/minecraft/world/entity/monster/Vindicator.java b/src/main/java/net/minecraft/world/entity/monster/Vindicator.java
-index ddf77350b1762be523eeed760669bc33f634109f..85294ab8eaa654be9358f489cec7e1ae665fc34b 100644
+index ddf77350b1762be523eeed760669bc33f634109f..d8bd9be2a7d2ce5cbbf39373cea1d1d9294c5354 100644
 --- a/src/main/java/net/minecraft/world/entity/monster/Vindicator.java
 +++ b/src/main/java/net/minecraft/world/entity/monster/Vindicator.java
 @@ -195,7 +195,7 @@ public class Vindicator extends AbstractIllager {
  
      static class VindicatorBreakDoorGoal extends BreakDoorGoal {
          public VindicatorBreakDoorGoal(Mob mob) {
 -            super(mob, 6, Vindicator.DOOR_BREAKING_PREDICATE);
-+            super(mob, 6, com.google.common.base.Predicates.in(mob.level.paperConfig.vindicatorBreakDoors)); // Paper
++            super(mob, 6, com.google.common.base.Predicates.in(mob.level.paperConfig.entitiesDifficultyBreakDoors.get(mob.getType()))); // Paper
```

Comment:
I'm actually not sure if its better to do mob.getType here, or EntityType.VINDICATOR. I'm thinking mob.getType might return some mob that doesnt have a value in the map which would make this null.

Reply:
yeah.. maybe can add the param for a default value if is missing using the default difficulty of vindicator